draft-ietf-ppm-dap.md

```diff
@@ -1514,217 +1609,234 @@ to the next phase of the aggregation protocol.
+} PrepareStep;
 ~~~
 
-where `[leader_prep_share, helper_prep_share]` is a vector of two elements. If
-either of these operations fails, then the leader marks the report as invalid
-with error "prep_share_failed". Otherwise it interprets `out` as follows: If
-this is the last round of the VDAF, then `out` is the Leader's output share, in
-which case it stores the output share for further processing as described in
-{{collect-flow}}. Otherwise, `out` is the pair `(next_prep_state,
-next_prep_share)`, where `next_prep_state` is its updated state and
-`next_prep_share` is its next preparation message-share (which will be
-`leader_prep_share` in the next round of continuation). For the latter case, the
-helper sets `prep_state` to `next_prep_state`.
+where `report_id` is the report ID associated with `state` and `outbound`.
 
-The Leader now advances its aggregation job to the next round (round 1 if this
-is the first continuation after initialization) and then instructs the Helper to
-advance the aggregation job to the round the Leader has just reached by sending
-the new `prep_msg` message to the Helper in a POST request to the aggregation
-job URI used during initialization (see {{leader-init}}). The body of the
-request is an `AggregationJobContinueReq`:
+Next, the Leader sends a POST request to the aggregation job URI used during
+initialization (see {{leader-init}}) with media type
+"application/dap-aggregation-job-continue-req" and body structured as:
 
 ~~~
 struct {
-  u16 round;
+  u16 step;
   PrepareStep prepare_steps<1..2^32-1>;
 } AggregationJobContinueReq;
 ~~~
 
-The `round` field is the round of VDAF preparation that the Leader just reached
-and wants the Helper to advance to.
+The `step` field is the step of DAP aggregation that the Leader just reached
+and wants the Helper to advance to. The `prepare_steps` field is the sequence
+of prep steps constructed in the previous step. The `PrepareStep`s MUST be in
+the same order as the previous aggregate request.
 
-The `prepare_steps` field MUST be a sequence of `PrepareStep`s in the
-`continued` state containing the corresponding `inbound` prepare message.
+The Helper's response will be an `AggregationJobResp` message (see
+{{aggregation-helper-init}}), which the Leader validates according to the
+criteria in {{aggregation-job-validation}}. If the message is invalid, the
+Leader MUST abort the aggregation job.
 
-The media type is set to "application/dap-aggregation-job-continue-req".
+Otherwise, the Leader proceeds as follows with each report:
 
-The Helper's response will be an `AggregationJobResp` message (see
-{{aggregation-helper-init}}), which the Leader validates according to
-the criteria in {{aggregation-job-validation}}. If the message is valid, the
-Leader moves to the next round of continuation with the enclosed prepare steps.
-Otherwise, the Leader should abandon the aggregation job entirely.
+1. If the inbound prep step type is "continue", then the Leader computes
```

Comment:
I think the condition here should be `inbound is "continue" && state == Continued(prep_state)`. If the Leader is instate `Finished`, then inbound prep step type being `"continue"` is a Helper-side bug and we should abort.